components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/gateway/v1/impl/ApiArtifactApiServiceImpl.java

```diff
@@ -18,24 +18,101 @@
+                            (gatewayAPIDTO.getTenantDomain());
+                    for (GatewayContentDTO gatewayEndpoint : gatewayAPIDTO.getEndpointEntriesToBeAdd()) {
+                        if (endpointAdminServiceProxy.isEndpointExist(gatewayEndpoint.getName())) {
+                            endPointArray.put(endpointAdminServiceProxy.getEndpoints(gatewayEndpoint.getName()));
+                        } else {
+                            log.error(gatewayEndpoint.getName() + " was not deployed in the gateway");
+                        }
+                    }
+                    responseObj.put("Endpoints", endPointArray);
+                }
+
+                if (gatewayAPIDTO.getLocalEntriesToBeAdd() != null) {
+                    JSONArray localEntryArray = new JSONArray();
+                    LocalEntryServiceProxy localEntryServiceProxy = new
+                            LocalEntryServiceProxy(gatewayAPIDTO.getTenantDomain());
+                    for (GatewayContentDTO localEntry : gatewayAPIDTO.getLocalEntriesToBeAdd()) {
+                        if (localEntryServiceProxy.isEntryExists(localEntry.getName())) {
+                            localEntryArray.put(localEntryServiceProxy.getEntry(localEntry.getName()));
+                        } else {
+                            log.error(localEntry.getName() + " was not deployed in the gateway");
+                        }
+                    }
+                    responseObj.put("Local Entries", localEntryArray);
+                }
+
+                if (gatewayAPIDTO.getSequenceToBeAdd() != null ) {
+                    JSONArray sequencesArray = new JSONArray();
+                    SequenceAdminServiceProxy sequenceAdminServiceProxy =
+                            new SequenceAdminServiceProxy(gatewayAPIDTO.getTenantDomain());
+                    for (GatewayContentDTO sequence : gatewayAPIDTO.getSequenceToBeAdd()) {
+                        if(sequenceAdminServiceProxy.isExistingSequence(sequence.getName())) {
+                            sequencesArray.put(sequenceAdminServiceProxy.getSequence(sequence.getName()));
+                        } else {
+                            log.error(sequence.getName() + " was not deployed in the gateway");
+                        }
+                    }
+                    responseObj.put("Sequences", sequencesArray);
+                }
+            } catch (EndpointAdminException e) {
+                log.error("Error in fetching deployed Endpoints from Synapse Configuration." , e);
+            } catch (AxisFault axisFault) {
+                log.error("Error in fetching deployed artifacts from Synapse Configuration." , axisFault);
+            }
+
             responseObj.put("Definition", definition);
             String responseStringObj = String.valueOf(responseObj);
             return Response.ok().entity(responseStringObj).build();
```

Comment:
we have to handle the errors too. If there was an error, need that details in the response